pkg/defaultcomponents/defaults.go

```diff
@@ -68,21 +68,24 @@ import (
 func Components() (otelcol.Factories, error) {
 	var errs error
 
-	extensions, err := extension.MakeFactoryMap(
+	var extensionsList = []extension.Factory{
```

Comment:
```suggestion
	extensionsList := []extension.Factory{
```

Any reason not to do short inline declaration here (and the other places in this file)?